src/containers/markdown/Files.js

```diff
@@ -0,0 +1,45 @@
+import React, { PureComponent } from 'react';
+import File from '../../components/File';
+import store from '../../store';
+import { getFiles } from '../../selectors/document';
+import { updateSelectedFile } from '../../actions/document';
+
+export default class Files extends PureComponent {
+  state = {
+    files: [],
+    selectedFile: 0
+  };
+
+  componentDidMount() {
+    this.setState({ files: getFiles(store.getState()) });
+    this.unsubscribe = store.subscribe(() => {
+      this.setState({ files: getFiles(store.getState()) });
+      console.log('state from store', store.getState());
+    });
+  }
+
+  componentWillUnmount() {
+    this.unsubscribe();
+  }
+
+  updateSelectedFile = ({ target }) => {
+    store.dispatch(updateSelectedFile(target.id));
+  };
+
+  render() {
+    const { files } = this.state;
+
+    const filesList = files.map(file => {
```

Comment:
I would extract this file list into a Files presentational component that renders the child File component.